src/item.cpp

```diff
@@ -1025,8 +1034,28 @@ bool item::stacks_with( const item &rhs, bool check_components ) const
     if( active != rhs.active ) {
         return false;
     }
-    if( item_tags != rhs.item_tags ) {
-        return false;
+    if( ignore_temperature && has_temperature() ) {
+        //compare ranges between "COLD", "FROZEN", "HOT".
+        if( !std::equal( rhs.item_tags.begin(), rhs.item_tags.lower_bound( flag_COLD ),
+                         item_tags.begin(), item_tags.lower_bound( flag_COLD ) ) ) {
+            return false;
+        }
+        if( !std::equal( rhs.item_tags.upper_bound( flag_COLD ), rhs.item_tags.lower_bound( flag_FROZEN ),
+                         item_tags.upper_bound( flag_COLD ), item_tags.lower_bound( flag_FROZEN ) ) ) {
+            return false;
+        }
+        if( !std::equal( rhs.item_tags.upper_bound( flag_FROZEN ), rhs.item_tags.lower_bound( flag_HOT ),
+                         item_tags.upper_bound( flag_FROZEN ), item_tags.lower_bound( flag_HOT ) ) ) {
+            return false;
+        }
+        if( !std::equal( rhs.item_tags.upper_bound( flag_HOT ), rhs.item_tags.end(),
+                         item_tags.upper_bound( flag_HOT ), item_tags.end() ) ) {
+            return false;
+        }
```

Comment:
If I understand correctly, you are comparing all these ranges of tags, specifically just to exclude the `COLD`, `FROZEN`, and `HOT` tags from making the items non-stackable when `ignore_temperature` is true. This not an efficient or robust way to compare/exclude tags; it would be more readable to filter those tags out before comparing. Again, I strongly recommend adding unit tests for this significant change in behavior.

However, there is a problem - remember, this kind of stacking should never happen with solid foods. With your change above, solid foods *will* stack together regardless of temperature, so a "microwave burrito (frozen)" will combine with a "microwave burrito (hot)" when placed in the same container, to make a "micorwave burrito (fresh) (2)". This is a regression. Only liquids can be combined / stacked at different temperatures. You don't have to handle the "is liquid" logic here, but it needs to be handled somewhere (probably in your new `item::can_combine` function).

Reply:
Range compare is efficient. The old `item_tags != rhs.item_tags` is to compare the two whole vectors. Vector and range compare are both loop of all elements. In order to "filter those tags out before comparing", we have to copy the two vectors. So, range compare in place is more efficient.

Reply:
I just tested without my patch, this kind of stacking does happen with solid foods. The old code does not check liquid. I didn't add or remove liquid checking. May I fix it in this PR or a new one?

Reply:
> I just tested without my patch, this kind of stacking does happen with solid foods. The old code does not check liquid. I didn't add or remove liquid checking. May I fix it in this PR or a new one?

I see now, hot and cold foods do stack before this patch as well. It would be great if you can fix that in this PR, since it's the same sort of bug (from user perspective) as flours and fruits stacking when they shouldn't.